pkg/apis/backupdriver/v1/backup_repository.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+/*
+ BackupRepository is a cluster-scoped resource.  It is controlled by the Backup Driver and referenced by
+ Snapshot, CloneFromSnapshot and Delete.  The BackupRespository resource contains the credential for a backup repository.
+ The RepositoryDriver defines the driver that will be used to talk to the repository
+ Only Snapshot,etc. CRs from namespaces that are listed in AllowedNamespaces will be acted on, if the namespace is
+ not in AllowedNamespaces the operation will fail.
+*/
+type BackupRepository struct {
+	// TypeMeta is the metadata for the resource, like kind and apiversion
+	meta_v1.TypeMeta `json:",inline"`
+
+	// +optional
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+
+	AllowedNamespaces []string `json:"allowedNamespaces"`
+
+	RepositoryDriver      string            `json:"repositoryDriver"`
+	Credential            map[string]string `json:"credential"`
```

Comment:
Credential - Is it just credentials or Config information as well like url, region, bucket name etc?

Reply:
I will change Credential to Parameters.  One of those parameters can be a secret reference, but secrets are really a construct for getting keys into a Pod, they don't seem to offer much advantage if you're reading/writing CRDs.  Secrets are usually just stored in Base64 encoding and the K8S privileges is relied on to prevent access.